spec/station12/reservations_controller_spec.rb

```diff
@@ -1,112 +1,66 @@
+    let(:no_date_request) { get :new, params: { movie_id: movie.id, schedule_id: schedule.id, sheet_id: sheets.first.id }, session: {} }
+    let(:no_sheet_request) { get :new, params: { movie_id: movie.id, schedule_id: schedule.id, date: date }, session: {} }
+
+    context "クエリについて" do
+      it 'date, sheet_idの両方があるときだけ200を返すこと' do
+        success_request
+        expect(response).to have_http_status(200)
+      end
+
+      it 'date, sheet_idの両方が渡されていないとき200を返していないこと' do
+        no_date_and_sheet_request
+        expect(response).not_to have_http_status(200)
+      end
+
+      it 'dateが渡されていないとき200を返していないこと' do
+        no_date_request
+        expect(response).not_to have_http_status(200)
+      end
+
+      it 'sheet_idが渡されていないとき200を返していないこと' do
+        no_sheet_request
+        expect(response).not_to have_http_status(200)
+      end
     end
 
-    it ' 200が返ること' do
-      expect(response).to have_http_status(200)
-    end
-
-    it '予約を全件出力していること' do
-      expect(response.body).to include(@first_reservation.name).and include(@first_reservation.name)
-      expect(response.body).to include(@last_reservation.email).and include(@last_reservation.email)
-    end
-  end
-
-  describe 'Station12 GET /admin/reservations/new' do
-    before { get :new }
-
-    it '200が返ること' do
-      expect(response).to have_http_status(200)
-    end
-
-    it 'name, email, schedule_id, sheet_idのすべてを受け取るフォームがあること' do
+    it 'form送信時にmovie_id, schedule_id, sheet_id, name, emailのすべてを送信するようになっていること' do
+      success_request
       expect(response.body).to include("name")
       expect(response.body).to include("email")
       expect(response.body).to include("schedule_id")
       expect(response.body).to include("sheet_id")
     end
   end
 
-  describe 'Station12 POST /admin/reservations/' do
-    it 'schedule_id, sheet_id, name, email, dateのすべてがあるときに302を返す' do
-      movie = create(:movie)
-      sheet = create(:sheet)
-      schedule = create(:schedule, movie_id: movie.id)
-      post :create, params: { reservation: { name: "TEST_NAME", email: "test@test.com", date: "2019-04-16", sheet_id: sheet.id , schedule_id: schedule.id, movie_id: movie.id }}, session: {}
+  describe 'Station12 POST /reservation/' do
+    let(:movie) { create(:movie) }
+    let(:sheets) { create_list(:sheet, 5) }
+    let(:date) { "2023-01-31" }
+    let(:schedule) { create(:schedule, movie_id: movie.id) }
+    let(:success_request) { post :create, params: { reservation: { name: "TEST_NAME", email: "test@test.com", date: date, sheet_id: sheets.first.id , schedule_id: schedule.id, movie_id: movie.id }}, session: {} }
 
+    it 'schedule_id, sheet_id, name, email, dateのすべてがあるときに302を返す' do
+      success_request
       expect(response).to have_http_status(302)
     end
 
-    it 'DBのunique制約にあたったときなどは400を返すこと' do
-      movie = create(:movie)
-      sheet = create(:sheet)
-      schedule = create(:schedule, movie_id: movie.id)
+    it 'DBのunique制約にかかったときにリダイレクトテスト' do
```

Comment:
```suggestion
    it 'DBのunique制約にかかったときにリダイレクトされること' do
```

テストであることはわかるため、何を確認するのかを記述すること